### PR TITLE
[Doc] Add documentation for parallel troubleshooting

### PR DESCRIPTION
**Pull request by rafaelbernard**
Add documentation for parallel troubleshooting and configuration.

Based on troubleshooting for issues https://github.com/rectorphp/rector/issues/7323 and https://github.com/rectorphp/rector/issues/7322

### CODE REVIEW

File: build/target-repository/docs/how_to_troubleshoot_parallel_issues.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+When that happens, the first good approach is to disable parallel processing. The output of parallel failures can hide some fatal errors from the source codebase. You can do so by:
+
+```php
+$rectorConfig->disableParallel();
+```
+
+After that, if rector processing works fine, that is an indication that you might need to adjust your parallel process to some balanced load, depending on the resources you have to process rector.
+
+
+[parallel() function has some defaults](https://github.com/rectorphp/rector-src/blob/main/packages/Config/RectorConfig.php#L53)
```
Comment (samsonasik) [inline build/target-repository/docs/how_to_troubleshoot_parallel_issues.md]: function -> method

File: build/target-repository/docs/how_to_troubleshoot_parallel_issues.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# How to Troubleshoot Parallel Issues
+
+You may get some parallel errors and ask how you can identify if there is something in your code or a real parallel issue than can be fixed by configuration. They may be displayed as:
+
+```
```
Comment (samsonasik) [inline build/target-repository/docs/how_to_troubleshoot_parallel_issues.md]:
```suggestion
```bash
```

File: build/target-repository/docs/how_to_troubleshoot_parallel_issues.md (hunk 1 above)
Comment (rafaelbernard) [inline build/target-repository/docs/how_to_troubleshoot_parallel_issues.md]:
```suggestion
If we check [parallel() method defaults](https://github.com/rectorphp/rector-src/blob/main/packages/Config/RectorConfig.php#L53):
```

File: build/target-repository/docs/how_to_troubleshoot_parallel_issues.md (hunk 2 above)
Comment (rafaelbernard) [inline build/target-repository/docs/how_to_troubleshoot_parallel_issues.md]:
```suggestion
```bash
```

**Comment by rafaelbernard** [comment]
Thanks for the review, @samsonasik . I applied the suggestions.

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Comment by rafaelbernard** [comment]
Thank you for the opportunity to start collaborating.

**Pull request merged**